packages/mobile_app/lib/components/user_settings/user_settings.dart

```diff
@@ -0,0 +1,166 @@
+import 'index.dart';
+
+Column userSettingsList({
+  @required MutableString name,
+  @required MutableString month,
+  @required MutableString day,
+  @required MutableString year,
+  @required MutableInteger sex,
+  @required Map<int, String> indexToSex,
+  @required MutableString country,
+  @required MutableString place,
+
+  /// defaultIsNull
+  MutableBool termsAccepted,
+  Function onUnvalidTerms,
+  //
+  @required Function onUnvalidInformation,
+  @required Function onValidInformation,
+  @required String buttonText,
+}) =>
+    Column(
+      children: <Widget>[
+        userInfo(
+          name: UserInfoField(
+              textObject: name,
+              maxLength: 100,
+              hint: lang.name.capitalize(),
+              validator: (String text) {
+                int min = 2;
+                if (text.isEmpty || text.length < min)
+                  return "${lang.atLeastXsymbolsNeeded} $min";
+                return null;
+              }),
+          month: UserInfoField(
+              textObject: month,
+              maxLength: 2,
+              keyboardType: TextInputType.datetime,
+              hint: "mm",
+              validator: (String text) {
+                int min = 2;
+                if (text.isEmpty || text.length < min)
+                  return "${text.length}/$min";
+                if (int.parse(text) > 12 || int.parse(text) < 1) return "x";
+                return null;
+              }),
+          day: UserInfoField(
+              textObject: day,
+              maxLength: 2,
+              keyboardType: TextInputType.datetime,
+              hint: "dd",
+              validator: (String text) {
+                int min = 2;
```

Comment:
User might want to enter `1` instead of `01`. Can we allow them to do so?

Reply:
I need to test it